File: packages/pigeon/example/native_interop_app/pigeons/native_interop_example.dart
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

import 'package:pigeon/pigeon.dart';

// #docregion config
@ConfigurePigeon(
  PigeonOptions(
    dartOptions: DartOptions(),
    kotlinOptions: KotlinOptions(
      useJni: true,
      // Optional: Paths to search for compiled local classes (primarily needed for standalone Apps)
      jniClassPaths: <String>['build/app/tmp/kotlin-classes/release'],
    ),
    swiftOptions: SwiftOptions(useFfi: true, ffiModuleName: 'Runner'),
  ),
)
// #enddocregion config
@HostApi()
abstract class NativeInteropExampleApi {
  void doSomething();
}

File: packages/pigeon/example/native_interop_app/android/app/src/main/kotlin/dev/flutter/pigeon_example_app/NativeInteropExample.kt
// Copyright 2013 The Flutter Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package dev.flutter.pigeon_example_app

// #docregion callback-style
fun echoAsync(value: String, callback: (Result<String>) -> Unit) {
  callback(Result.success(value))
}
// #enddocregion callback-style

// #docregion concurrency-style
suspend fun echoAsync(value: String): String {
  return value
}
// #enddocregion concurrency-style
